src/mnist_loader.py

```diff
@@ -35,24 +44,40 @@ def load_fashion_mnist():
         print("No fashion MNIST test (10k) images found--downloading")
         _download_file("http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-images-idx3-ubyte.gz", 'data/fashion')
     print ('Loading testing images...')
-    t10k_images = _load_image("data/fashion/t10k-images-idx3-ubyte.gz")
+    test_images = _load_image("data/fashion/t10k-images-idx3-ubyte.gz")
 
     if not os.path.exists("data/fashion/t10k-labels-idx1-ubyte.gz"):
         print("No fashion MNIST test (10k) labels found--downloading")
         _download_file("http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-labels-idx1-ubyte.gz", 'data/fashion')
     print ('Loading testing labels...')
-    t10k_labels = _load_label('data/fashion/t10k-labels-idx1-ubyte.gz')
+    test_labels = _load_label('data/fashion/t10k-labels-idx1-ubyte.gz')
 
-    return train_images, train_labels, t10k_images, t10k_labels
+    if validation_proportion is None or validation_proportion == 0:
+        return train_images, train_labels, test_images, test_labels
+    else:
+        print('stop here')
+        (train_images, val_images,  train_labels, val_labels) = train_test_split(train_images, train_labels,
+                                                                                 random_state=validation_seed,
+                                                                                 test_size=validation_proportion)
+        return train_images, train_labels, val_images, val_labels, test_images, test_labels
 
 
-def load_mnist():
+def load_mnist(validation_proportion=None, validation_seed=None):
     """
     Get the MNIST training data (downloading it if it is not already accessible),
     and return it as NumPy arrays
 
-    :return: (train_images, train_labels, test_images, test_labels)
+
+    Args:
+        validation_proportion: If not None, extract this much data from the training set and return it as a separate
+            validation set
+        validation_seed: seed/state for selecting validation data
+
+
+    :return: (train_images, train_labels, test_images, test_labels) if validation_proportion=None or zero, otherwise
+        returns (trainimages, train_labels, val_images, val_labels, test_images, test_labels)
     """
+
```

Comment:
You should make a check that validation_proportion is between [0,1]

Reply:
train_test_split throws a ValueError if it's out of range already but I can add it if you want.

Reply:
that's alright then. no need to do the same check.